src/Database/Type/DecimalType.php

```diff
@@ -158,7 +158,12 @@ public function marshal($value)
             return (float)$value;
         }
 
-        return null;
+        // allow custom decimal format (@see https://github.com/cakephp/cakephp/issues/12800)
+        if (preg_match('/[^0-9,. ]/', $value)) {
+            return null;
```

Comment:
You have incorrectly swapped the return values :)

Reply:
Sure? The ^ means not afaik.

Reply:
I hope it's correct